incubator/hnc/pkg/forest/forest.go

```diff
@@ -51,70 +51,76 @@ func (f *Forest) AddOrGet(nm string) *Namespace {
 	return ns
 }
 
-// Get returns a `Namespace` object if it exists, or nil if it's missing.
-func (f *Forest) Get(nm string) *Namespace {
-	ns, _ := f.namespaces[nm]
-	return ns
-}
-
-// Remove deletes the given namespace and returns a list of all namespaces who
-// have been affected (ie, the parent and the now-orphaned children).
-func (f *Forest) Remove(nm string) []string {
-	affected := []string{}
-	ns, _ := f.namespaces[nm]
-	if ns == nil {
-		return nil
-	}
-
-	// Update the parent, if any
-	p := ns.Parent()
-	if p != nil {
-		affected = append(affected, p.name)
-		delete(p.children, nm)
-	}
-
-	// Unset the parent from all children, if any
-	for _, c := range ns.children {
-		affected = append(affected, c.name)
-		c.parent = nil
-	}
-
-	// Delete and return affected
-	delete(f.namespaces, nm)
-	return affected
-}
-
 type namedNamespaces map[string]*Namespace
 
-// Namespace represents a namespace in a forest. For now, its only properties are its own name, its
-// parent, and its children.
+// Namespace represents a namespace in a forest. Other than its structure, it contains some
+// properties useful to the reconcilers.
 type Namespace struct {
 	forest   *Forest
 	name     string
 	parent   *Namespace
 	children namedNamespaces
+	exists   bool
+}
+
+// Exists returns true if the namespace exists.
+func (ns *Namespace) Exists() bool {
+	return ns.exists
+}
+
+// Create marks this namespace as existing, returning true if didn't previously exist.
+func (ns *Namespace) Create() bool {
```

Comment:
Can we add an SetExists() interface and replace Create with 
If !ns.Exist() {
   ns.SetExist()
}
I feel this improves readability.

Reply:
Done. Also changed `Remove` to `UnsetExists`.